web3/_utils/method_formatters.py

```diff
@@ -569,6 +586,14 @@ def raise_solidity_error_on_revert(response: RPCResponse) -> RPCResponse:
         reason = data[len(prefix) + 64:len(prefix) + 64 + reason_length * 2]
         raise ContractLogicError(f'execution reverted: {bytes.fromhex(reason).decode("utf8")}')
 
+    # --- EIP-3668 | CCIP Read --- #
+    # 0x556f1830 is the function selector for OffchainLookup(address,string[],bytes,bytes4,bytes)
+    if data[:10] == '0x556f1830':
```

Comment:
This feels like it should be in it's own function since it's not raising a solidity error (i.e. a `ContractLogicError`). I propose we either rename this function or pull lines 591-596 out into their own function.

Reply:
Good point. Alternatively, should this maybe extend from `ContractLogicError`? It's technically an `error` in solidity, even though it's a purposeful error.

Reply:
Yeah, extending from `ContractLogicError` makes sense to me since it stems from a `revert` 👍